### PR TITLE
fix typos and other minor translation issues

### PR DESCRIPTION
**Pull request by swurzinger**
- fix typos in code, comments and translations
   - no changes in DB structure
- remove duplicate translation strings
- fix some DE/IT/FR/ES translations

There are still quite some errors in the original strings as well as translations, which are not fixed in this PR.

### CODE REVIEW

**Comment by swurzinger** [comment]
will rebase onto 4.8 after #1321 is merged

File: tests/Traits/WithExpenseTest.php [hunk 1]
```diff
@@ -98,7 +98,7 @@ protected function attemptCreateExpenses()
                     'value' => 1500,
                     'recurring' => false,
                     'category_id' => $category->id,
-                    'occurence' => 'month_starts',
+                    'occurrence' => 'month_starts',
```
Comment (Blair2004) [inline tests/Traits/WithExpenseTest.php:101]: A database migration is needed to update the existing installations (current users installation) that has this attribute on the database. Unless the migration is added to the PR, this should be skipped for now.

**Review by Blair2004 — CHANGES_REQUESTED** [review]
Hi,
I couldn't imagine there was that much typos... Your contribution is amazing.
Thank you so much.

There is a change need regarding the an attribute name which is currently persistent on users installation. We need to create a migration that update the column name on current installations.

File: tests/Traits/WithExpenseTest.php (hunk 1 above)
Comment (swurzinger) [reply tests/Traits/WithExpenseTest.php:101]: thanks, I reverted some other usages of `occurence` related to DB entities, but must have missed these ones. I guess fixing the DB column name as well as the name in data structures is not worth the effort as it's a breaking change for existing modules.

Comment (Blair2004) [reply tests/Traits/WithExpenseTest.php]: We'll ship this for v5 and suggest modules to update their implementation as well.

**Review by Blair2004 — APPROVED** [review]

**Pull request merged**